datalad/support/gitrepo.py

```diff
@@ -809,7 +809,7 @@ def pull(self, remote=None, refspec=None, progress=None, **kwargs):
             remote.pull(refspec=refspec, progress=progress, **kwargs)
             # TODO: progress +kwargs
 
-    def push(self, remote=None, refspec=None, progress=None, all=False, **kwargs):
+    def push(self, remote=None, refspec=None, progress=None, all_=False, **kwargs):
         """See fetch
         """
```

Comment:
just FYI that in some cases helper we have (from PyMVPA) `@borrowdoc`, but apparently not using it yet might be of help.